index.js

```diff
@@ -129,8 +129,31 @@ function pluralTypeName(pluralRules, locale) {
     || langToPluralType.en;
 }
 
-function pluralTypeIndex(pluralRules, locale, count) {
-  return pluralRules.pluralTypes[pluralTypeName(pluralRules, locale)](count);
+function pluralTypeIndex(pluralRules, pluralType, count) {
+  return pluralRules.pluralTypes[pluralType](count);
+}
+
+function createMemoizedPluralTypeNameSelector() {
+  var localePluralTypeStorage = {};
+
+  return function (pluralRules, locale) {
+    var pluralType = localePluralTypeStorage[locale];
+
+    if (pluralType && !pluralRules.pluralTypes[pluralType]) {
+      delete localePluralTypeStorage[locale];
```

Comment:
instead of deleting this - which deoptimizes and slows down objects - could it be set to null?